mypy cannot import types for package and dependencies

Install package and dependencies into mypy session. mypy needs to import the types for the package, as well as for the core and development dependencies. Revert an earlier commit where we removed this installation.

This problem only manifests in one of the following cases:

- the mypy command-line does not also include the package itself
- we rely on inline type information distributed with a third-party package

(In many small projects, simply running `nox` will not manifest the bug. But it's there.)

Example from poetry-merge-lock:

```
nox -rs mypy-3.8 -- tests/test_main.py
nox > Running session mypy-3.8
nox > Re-using existing virtual environment at .nox/mypy-3-8.
nox > poetry export --dev --format=requirements.txt --output=/var/folders/13/g258r36n3fd7rj0jrgln5gd04dw8n3/T/tmpo1_ho48l/requirements.txt
nox > pip install --constraint=/var/folders/13/g258r36n3fd7rj0jrgln5gd04dw8n3/T/tmpo1_ho48l/requirements.txt mypy
nox > mypy tests/test_main.py
tests/test_main.py:5:1: error: Cannot find implementation or library stub for module named 'poetry_merge_lock'  [import]
    from poetry_merge_lock import __main__
    ^
tests/test_main.py:5:1: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
tests/test_main.py:5:1: error: Cannot find implementation or library stub for module named 'poetry_merge_lock.__main__'  [import]
    from poetry_merge_lock import __main__
    ^
Found 2 errors in 1 file (checked 1 source file)
nox > Command mypy tests/test_main.py failed with exit code 1
nox > Session mypy-3.8 failed.
```

--- a/{{cookiecutter.project_name}}/noxfile.py
+++ b/{{cookiecutter.project_name}}/noxfile.py
@@ -131,6 +131,7 @@ def safety(session: Session) -> None:
 def mypy(session: Session) -> None:
     """Type-check using mypy."""
     args = session.posargs or locations
+    install_package(session)
     install(session, "mypy")
     session.run("mypy", *args)
 

